### PR TITLE
Remove now unused patches to ttconv.

### PR DESCRIPTION
**Pull request by anntzer**
get_pdf_charprocs and TTDictionaryCallback were added in 7a10b51.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by timhoffm — APPROVED** [review]

File: extern/ttconv/pprdrv_tt.cpp [hunk 1]
```diff
@@ -1417,22 +1417,6 @@ class StringStreamWriter : public TTStreamWriter
...
-void get_pdf_charprocs(const char *filename, std::vector<int>& glyph_ids, TTDictionaryCallback& dict)
-{
-    struct TTFONT font;
-
-    read_font(filename, PDF_TYPE_3, glyph_ids, font);
-
-    for (std::vector<int>::const_iterator i = glyph_ids.begin();
-            i != glyph_ids.end(); ++i)
-    {
-        StringStreamWriter writer;
```
Comment (QuLogic) [inline extern/ttconv/pprdrv_tt.cpp:1429]: The `StringStreamWriter` above this appears unused as well.

Comment (anntzer) [reply extern/ttconv/pprdrv_tt.cpp:1429]: indeed; I also removed all the pdf_mode-related patches which came in at the same time.

File: extern/ttconv/pprdrv_tt2.cpp [hunk 2]
```diff
@@ -464,50 +459,27 @@ void GlyphToType3::do_composite(TTStreamWriter& stream, struct TTFONT *font, BYT
...
-            }
+            stream.printf("%% unimplemented shift, arg1=%d, arg2=%d\n",arg1,arg2);
+        }
 
-            /* Invoke the CharStrings procedure to print the component. */
-            stream.printf("false CharStrings /%s get exec\n",
-                          ttfont_CharStrings_getname(font,glyphIndex));
+        /* Invoke the CharStrings procedure to print the component. */
+        stream.printf("false CharStrings /%s get exec\n",
+                        ttfont_CharStrings_getname(font,glyphIndex));
```
Comment (QuLogic) [inline extern/ttconv/pprdrv_tt2.cpp:476]:
```suggestion
                      ttfont_CharStrings_getname(font,glyphIndex));
```

**Review by QuLogic — APPROVED** [review]

File: extern/ttconv/pprdrv_tt2.cpp (hunk 2 above)
Comment (anntzer) [reply extern/ttconv/pprdrv_tt2.cpp]: yes

**Pull request merged**